metamapper/peakmapper.py

```diff
@@ -0,0 +1,156 @@
+import coreir
+from .metamapper import MetaMapper
+from collections import OrderedDict
+from hwtypes import BitVector, AbstractBit, AbstractBitVector
+from peak.mapper import gen_mapping, SMTBitVector
+import peak
+from .rewrite_rule import Peak1to1, PeakIO
+
+
+
+class PeakMapper(MetaMapper):
+    def __init__(self,context : coreir.context,namespace_name):
+        self.peak_primitives = {}
+        self.io_primitives = {}
+        self.discover_constraints = []
+        super(PeakMapper,self).__init__(context,namespace_name)
+ 
+    def extract_instr_map(self,app : coreir.module.Module):
```

Comment:
I'm confused. This function doesn't use the argument `app` at all. How is that?

Reply:
Right now I keep around the maps stored in the rewrite rule itself. I want to eventually move into adding the instruction as direct metadata on pycoreir objects, but that is NYI. https://github.com/leonardt/pycoreir/issues/79#issuecomment-479188460

Reply:
Yes, that is true. This is a bit of a hack currently.